apis/apiextensions/v2alpha1/xrd_types.go

```diff
@@ -0,0 +1,314 @@
+	DeprecationWarning *string `json:"deprecationWarning,omitempty"`
+
+	// Schema describes the schema used for validation, pruning, and defaulting
+	// of this version of the defined composite resource. Fields required by all
+	// composite resources will be injected into this schema automatically, and
+	// will override equivalently named fields in this schema. Omitting this
+	// schema results in a schema that contains only the fields required by all
+	// composite resources.
+	// +optional
+	Schema *CompositeResourceValidation `json:"schema,omitempty"`
+
+	// AdditionalPrinterColumns specifies additional columns returned in Table
+	// output. If no columns are specified, a single column displaying the age
+	// of the custom resource is used. See the following link for details:
+	// https://kubernetes.io/docs/reference/using-api/api-concepts/#receiving-resources-as-tables
+	// +optional
+	AdditionalPrinterColumns []extv1.CustomResourceColumnDefinition `json:"additionalPrinterColumns,omitempty"`
+}
+
+// CompositeResourceValidation is a list of validation methods for a composite
+// resource.
+type CompositeResourceValidation struct {
+	// OpenAPIV3Schema is the OpenAPI v3 schema to use for validation and
+	// pruning.
+	// +kubebuilder:pruning:PreserveUnknownFields
+	OpenAPIV3Schema runtime.RawExtension `json:"openAPIV3Schema,omitempty"` //nolint:tagliatelle // False positive. Linter thinks it should be Apiv3, not APIV3.
+}
+
+// CompositeResourceDefinitionStatus shows the observed state of the definition.
+type CompositeResourceDefinitionStatus struct {
+	xpv1.ConditionedStatus `json:",inline"`
+
+	// Controllers represents the status of the controllers that power this
+	// composite resource definition.
+	Controllers CompositeResourceDefinitionControllerStatus `json:"controllers,omitempty"`
+}
+
+// CompositeResourceDefinitionControllerStatus shows the observed state of the
+// controllers that power the definition.
+type CompositeResourceDefinitionControllerStatus struct {
+	// The CompositeResourceTypeRef is the type of composite resource that
+	// Crossplane is currently reconciling for this definition. Its version will
+	// eventually become consistent with the definition's referenceable version.
+	// Note that clients may interact with any served type; this is simply the
+	// type that Crossplane interacts with.
+	CompositeResourceTypeRef TypeReference `json:"compositeResourceType,omitempty"`
+
+	// The CompositeResourceClaimTypeRef is the type of composite resource claim
+	// that Crossplane is currently reconciling for this definition. Its version
+	// will eventually become consistent with the definition's referenceable
+	// version. Note that clients may interact with any served type; this is
+	// simply the type that Crossplane interacts with.
+	CompositeResourceClaimTypeRef TypeReference `json:"compositeResourceClaimType,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +genclient
+// +genclient:nonNamespaced
```

Comment:
what does it mean for comment markers to not be directly on any type, does it apply it to all remaining types in the file?

Reply:
I think it only applies to `CompositeResourceDefinition` in this case. Not exactly sure how that association is formed though - maybe just the next type seen after the comment?

FWIW In this case this was just copied from the v1 equivalent file.